### PR TITLE
Upgrade to richie 2.0.0 beta.20

### PR DESCRIPTION
**Pull request by jbpenrath**
## Purpose

Upgrade sites to richie 2.0.0 beta.20

The new release of richie-education includes:

-  frontend translation overriding,

-  frontend compilation with webpack 5 

- section title customization

- Django > 3 with Django CMS 3.8

- SSO

- Color palettes for spinner & selector components

This PR depends on https://github.com/openfun/richie/pull/1162

## Proposal
- [x] Preconfigure each sites to support frontend translation overriding
- [x] Migrate frontend compilation commands to webpack 5
- [x] Update styles to support section title customization
- [x] Update apps settings
- [x] Migrate to Django > 3
- [x] Add color palettes for spinner & selector for each sites

### CODE REVIEW

File: sites/funcorporate/src/frontend/package.json [hunk 1]
```diff
@@ -31,10 +33,10 @@
   },
   "homepage": "https://github.com/openfun/richie-site-factory#readme",
   "dependencies": {
-    "react": "16.13.1",
-    "richie-education": "2.0.0-beta.15"
+    "@hi4s/richie-education": "2.0.0-beta.16-6"
```
Comment (mbenadda) [inline sites/funcorporate/src/frontend/package.json:36]: Not sure you wanted to commit this

File: .circleci/config.yml [hunk 2]
```diff
@@ -333,47 +334,322 @@ version: 2.1
 workflows:
     cnfpt:
         jobs:
-            - no-change:
```
Comment (mbenadda) [inline .circleci/config.yml:336]: Not sure how all these jobs changes are relevant to translations? Maybe this deserves a mention in commit messages at least if it is intended

File: Makefile [hunk 3]
```diff
@@ -217,12 +217,22 @@ import-fixtures:  ## import fixtures
...
+	i18n-compile-front
+.PHONY: i18n
+
+i18n-compile-back: ## create/update .po files and compile .mo files used for i18n
 	@$(MANAGE) makemessages --keep-pot
 	@echo 'Reactivating obsolete strings (allow overriding strings defined in dependencies)'
 	@$(COMPOSE_RUN_APP) find ./ -type f -name django.po -exec sed -i 's/#~ //g' {} \;
 	@$(MANAGE) compilemessages
-.PHONY: i18n
+.PHONY: i18n-compile-back
```
Comment (sampaccoud) [inline Makefile:231]: should be just "i18n-back" as we do extraction and compilation in the same command (I chose to group everything in one command to simplify things in rsf).

Maybe do the same for frontend translations?

File: sites/demo/src/frontend/package.json [hunk 4]
```diff
@@ -31,9 +33,10 @@
   },
   "homepage": "https://github.com/openfun/richie-site-factory#readme",
   "dependencies": {
-    "richie-education": "2.0.0-beta.15"
+    "richie-education": "2.0.0-beta.17"
```
Comment (sampaccoud) [inline sites/demo/src/frontend/package.json:36]: You need to add a line in each CHANGELOG.md file mentioning the upgrade
 (this is not taken care of by the bin/upgrade script yet and must be done manually...)

File: sites/cnfpt/src/backend/cnfpt/settings.py [hunk 5]
```diff
@@ -138,6 +138,19 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     # Security
     ALLOWED_HOSTS = values.ListValue([])
     SECRET_KEY = "ThisIsAnExampleKeyForDevPurposeOnly"  # nosec
+    # System check reference:
+    # https://docs.djangoproject.com/en/2.2/ref/checks/#security
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/cnfpt/settings.py:142]:
```suggestion
    # https://docs.djangoproject.com/en/3.1/ref/checks/#security
```

File: sites/cnfpt/src/backend/cnfpt/settings.py [hunk 6]
```diff
@@ -192,20 +205,49 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     SOCIAL_AUTH_EDX_OAUTH2_ENDPOINT = values.Value()
     SOCIAL_AUTH_POSTGRES_JSONFIELD = False
 
+    # AUTHENTICATION
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/cnfpt/settings.py]: Should be under "# LMS" as this is not really authentication for the richie app...

File: sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 7]
```diff
@@ -21,24 +21,40 @@ msgstr ""
...
+msgstr ""
+
+#: cnfpt/settings.py:226
+msgid "Account"
+msgstr ""
+
+#: cnfpt/settings.py:226
+msgid "{base_url:s}/account/settings"
+msgstr ""
+
```
Comment (sampaccoud) [inline sites/cnfpt/src/backend/locale/fr/LC_MESSAGES/django.po:39]: did you forget the translations?

File: sites/demo/src/backend/demo/settings.py [hunk 8]
```diff
@@ -142,6 +142,19 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     # Security
     ALLOWED_HOSTS = values.ListValue([])
     SECRET_KEY = "ThisIsAnExampleKeyForDevPurposeOnly"  # nosec
+    # System check reference:
+    # https://docs.djangoproject.com/en/2.2/ref/checks/#security
```
Comment (sampaccoud) [inline sites/demo/src/backend/demo/settings.py:146]:
```suggestion
    # https://docs.djangoproject.com/en/3.1/ref/checks/#security
```

File: sites/demo/src/backend/demo/settings.py [hunk 9]
```diff
@@ -229,20 +242,49 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
         "social_core.pipeline.user.user_details",
     )
 
+    # AUTHENTICATION
```
Comment (sampaccoud) [inline sites/demo/src/backend/demo/settings.py:245]: ditto

File: sites/funmooc/src/backend/funmooc/settings.py [hunk 10]
```diff
@@ -186,6 +199,54 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     LOGIN_REDIRECT_URL = "/"
     LOGOUT_REDIRECT_URL = "/"
 
+    # AUTHENTICATION
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/funmooc/settings.py:202]: ditto

File: sites/funmooc/src/backend/funmooc/settings.py [hunk 11]
```diff
@@ -143,6 +143,19 @@ class Base(StyleguideMixin, DRFMixin, RichieCoursesConfigurationMixin, Configura
     # Security
     ALLOWED_HOSTS = values.ListValue([])
     SECRET_KEY = "ThisIsAnExampleKeyForDevPurposeOnly"  # nosec
+    # System check reference:
+    # https://docs.djangoproject.com/en/2.2/ref/checks/#security
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/funmooc/settings.py:147]:
```suggestion
    # https://docs.djangoproject.com/en/3.1/ref/checks/#security
```

File: sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po [hunk 12]
```diff
@@ -21,11 +21,27 @@ msgstr ""
...
+msgstr ""
+
+#: funmooc/settings.py:220
+msgid "Account"
+msgstr ""
+
+#: funmooc/settings.py:220
+msgid "{base_url:s}/account/settings"
+msgstr ""
+
```
Comment (sampaccoud) [inline sites/funmooc/src/backend/locale/fr/LC_MESSAGES/django.po:39]: You should translated these strings in French

File: sites/cnfpt/src/backend/cnfpt/settings.py (hunk 5 above)
Comment (jbpenrath) [reply sites/cnfpt/src/backend/cnfpt/settings.py:142]: Good catch!

**Review by sampaccoud — APPROVED** [review]

**Pull request merged**